src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1509,3 +1636,54 @@ void java_bytecode_parsert::parse_local_variable_type_table(methodt &method)
       "Entry in LocalVariableTypeTable must be present in LVT");
   }
 }
+
+/// Read method handle pointed to from constant pool entry at index, return type
+/// of method handle and name if lambda function is found.
+/// \param entry: the constant pool entry of the methodhandle_info structure
+/// \param[out]: the method_handle type of the methodhandle_structure, either
+/// for a recognized bootstrap method, for a lambda function or unknown
+void java_bytecode_parsert::parse_methodhandle(
+  const pool_entryt &entry,
+  lambda_method_handlet &handle)
+{
+  INVARIANT(
+    entry.tag == CONSTANT_MethodHandle,
+    "constant pool entry must be a MethodHandle");
+  const auto &ref_entry = pool_entry(entry.ref2);
+  INVARIANT(
+    (entry.ref1 > 0 && entry.ref1 < 10),
+    "reference kind of Methodhandle must be in the range of 1 to 9");
+
+  const auto &class_entry = pool_entry(ref_entry.ref1);
+  const auto &nameandtype_entry = pool_entry(ref_entry.ref2);
+
+  const std::string method_name =
+    id2string(pool_entry(class_entry.ref1).s) + "."
+    + id2string(pool_entry(nameandtype_entry.ref1).s)
+    + id2string(pool_entry(nameandtype_entry.ref2).s);
+
+  if(
+    method_name ==
+    "java/lang/invoke/LambdaMetafactory.metafactory(Ljava/lang/invoke/"
+    "MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/invoke/MethodType;Ljava/"
+    "lang/invoke/MethodType;Ljava/lang/invoke/MethodHandle;Ljava/lang/invoke/"
+    "MethodType;)Ljava/lang/invoke/CallSite;")
+    handle.handle_type = method_handle_typet::BOOTSTRAP_METHOD_HANDLE;
+
+  // names seem to be lambda$$POSTFIX$NUM
+  // where $POSTFIX is $FUN for a function name in which the lambda is define
+  //                   "static" when it is a static member of the class
+  //                   "new" when it is a class variable, instantiated in <init>
+  if(has_prefix(id2string(pool_entry(nameandtype_entry.ref1).s), "lambda$"))
```

Comment:
else if.

Also remove linebreaks between end of `if` and start of `else if` to clarify that they belong together